config/config.go

```diff
@@ -67,26 +67,30 @@ type SourceBlockchain struct {
 	MessageContracts                  map[string]MessageProtocolConfig `mapstructure:"message-contracts" json:"message-contracts"`
 	SupportedDestinations             []string                         `mapstructure:"supported-destinations" json:"supported-destinations"`
 	ProcessHistoricalBlocksFromHeight uint64                           `mapstructure:"process-historical-blocks-from-height" json:"process-historical-blocks-from-height"`
+	AllowedOriginSenderAddresses      []string                         `mapstructure:"allowed-origin-sender-addresses" json:"allowed-origin-sender-addresses"`
 
 	// convenience fields to access parsed data after initialization
-	supportedDestinations set.Set[ids.ID]
-	subnetID              ids.ID
-	blockchainID          ids.ID
+	supportedDestinations        set.Set[ids.ID]
+	subnetID                     ids.ID
+	blockchainID                 ids.ID
+	allowedOriginSenderAddresses []common.Address
 }
 
 type DestinationBlockchain struct {
-	SubnetID          string `mapstructure:"subnet-id" json:"subnet-id"`
-	BlockchainID      string `mapstructure:"blockchain-id" json:"blockchain-id"`
-	VM                string `mapstructure:"vm" json:"vm"`
-	RPCEndpoint       string `mapstructure:"rpc-endpoint" json:"rpc-endpoint"`
-	AccountPrivateKey string `mapstructure:"account-private-key" json:"account-private-key"`
+	SubnetID                    string   `mapstructure:"subnet-id" json:"subnet-id"`
+	BlockchainID                string   `mapstructure:"blockchain-id" json:"blockchain-id"`
+	VM                          string   `mapstructure:"vm" json:"vm"`
+	RPCEndpoint                 string   `mapstructure:"rpc-endpoint" json:"rpc-endpoint"`
+	AccountPrivateKey           string   `mapstructure:"account-private-key" json:"account-private-key"`
+	AllowedDestinationAddresses []string `mapstructure:"allowed-destination-addresses" json:"allowed-destination-addresses"`
```

Comment:
Would it make more sense to define the allowed destinations as part of the `supportedDestinations` of the `SourceBlockchain` configuration? I think it's a bit confusing to have supported destinations defined separately from allowed destination addresses, and I think is not as specific as could be required.

For instance, if a relayer supports relaying messages bidirectionally between `A<->B`, and `A<->C`, then the allowed destination addresses defined for on Subnet A will apply for messages from both Subnets B and C, which doesn't map will into the intended application relayer instances that should be created.

I'd recommend defining the `SupportedDestinations` of a `SourceSubnet` fully along the lines of:
```
type SupportedDestination {
    BlockchainID ids.ID
    DestinationAddresses []common.Address
}
```

And then not having any additional configuration of the `DestinationBlockchain` struct.

Reply:
I went ahead and made this change; I think it makes a lot of sense. It also brings us one step closer to having a config where the application relayer configuration is standalone at the top level. I'm thinking something like:

`type Config struct {
    ApplicationRelayers []ApplicationRelayer
    SourceBlockchains []SourceBlockchain
    DestinationBlockchains []DestinationBlockchain
}
`

where `SourceBlockchain` and `DestinationBlockchain` specify how to connect to the chains, and `ApplicationRelayer` specifies a distinct relayer path.